uaclient/data_types.py

```diff
@@ -48,6 +61,25 @@ def from_value(val: Any) -> Any:
         return val
 
 
+class EnumDataValue(DataValue):
+    """
+    To be used for parsing enum values
+    from_value raises an error if the value is not in the enum class values
+    and returns the value if found.
+    """
+
+    @classmethod
+    def from_value(cls, val: Any) -> str:
+        expected_values = tuple(i.value for i in cls)  # type: ignore
+        if not isinstance(val, type(expected_values[0])):
+            raise IncorrectTypeError(
+                type(val).__name__, type(expected_values[0]).__name__
+            )
+        if val not in expected_values:
+            raise IncorrectEnumValueError(str(val), cls)
```

Comment:
I think we want to return an actual enum value here, not the raw int/string.

Also, turns out enums have a lot of what we need already! This can be simplified to something like:
```suggestion
        try:
            return = cls(val)
        except ValueError:
            raise IncorrectEnumValueError([i.value for i in cls])
```